front/app/containers/Admin/reporting/components/ReportBuilder/Widgets/SurveyResultsWidget/SurveyQuestionFilter.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { ChangeEvent } from 'react';
+
+// hooks
+import useProject from 'hooks/useProject';
+import useFormResults from 'hooks/useFormResults';
+import useLocalize from 'hooks/useLocalize';
+
+// utils
+import { useIntl } from 'utils/cl-intl';
+import { isNilOrError } from 'utils/helperUtils';
+import { Box, Checkbox, Text } from '@citizenlab/cl2-component-library';
+
+// messages
+import messages from './messages';
+
+type SurveyQuestionFilterProps = {
+  projectId: string;
+  phaseId: string;
+  showQuestions: number[] | undefined;
+  onToggleQuestion: (showQuestions: number[]) => void;
+};
+
+const SurveyQuestionFilter = ({
+  projectId,
+  phaseId,
+  showQuestions,
+  onToggleQuestion,
+}: SurveyQuestionFilterProps) => {
+  const localize = useLocalize();
+  const { formatMessage } = useIntl();
+  const project = useProject({ projectId });
+  const formResults = useFormResults({
+    projectId,
+    phaseId,
+  });
+
+  if (isNilOrError(formResults) || isNilOrError(project)) {
+    // This never seems to get shown
```

Comment:
If this never gets shown and it's just a type guard, it should probably return `null`. But I guess it is possible that you're showing a survey without multiple choice questions, so then this should be rendered, right? What does the backend do in that case?

Reply:
Well, I assume there is a chance that a project ID could get passed that has saved but has since been deleted, which would throw an error. Not sure about form results.